src/utils/augmentation.py

```diff
@@ -0,0 +1,132 @@
+import random
+import numpy as np
+
+from abc import abstractmethod, ABC
+
+
+class AugmentationWord(ABC):
+    @abstractmethod
+    def __call__(self, *args, **kwargs):
+        """
+        Generates a new noisy word
+        """
+
+        raise NotImplementedError
+
+
+class SwapAugmentation(AugmentationWord):
+    """Swap one (random) pair of adjacent characters in a given word"""
+
+    def __init__(self):
+        pass
+
+    def __call__(self, word: str, proba_per_char: float) -> str:
```

Comment:
`proba_per_char` should be passed in augmentation constructor. Inside this method you would use `sellf.proba_per_char`

Reply:
Btw, you don't use this proba, so no need to pass it to this class anyway